tests/components/inels/test_switch.py

```diff
@@ -0,0 +1,167 @@
+from homeassistant.core import HomeAssistant
+
+from .common import MAC_ADDRESS, UNIQUE_ID, get_entity_id
+
+DT_07 = "07"
+DT_100 = "100"
+DT_BITS = "bits"
+
+
+@pytest.fixture(params=["simple_relay", "relay", "bit"])
+def entity_config(request: pytest.FixtureRequest):
+    """Fixture to provide parameterized entity configuration for switch tests."""
+    configs = {
+        "simple_relay": {
+            "entity_type": "switch",
+            "device_type": "simple_relay",
+            "dev_type": DT_07,
+            "unique_id": UNIQUE_ID,
+            "gw_connected_topic": f"inels/connected/{MAC_ADDRESS}/gw",
+            "connected_topic": f"inels/connected/{MAC_ADDRESS}/{DT_07}/{UNIQUE_ID}",
+            "status_topic": f"inels/status/{MAC_ADDRESS}/{DT_07}/{UNIQUE_ID}",
+            "base_topic": f"{MAC_ADDRESS}/{DT_07}/{UNIQUE_ID}",
+            "switch_on_value": "07\n01\n92\n09\n",
+            "switch_off_value": "07\n00\n92\n09\n",
+        },
+        "relay": {
+            "entity_type": "switch",
+            "device_type": "relay",
+            "dev_type": DT_100,
+            "unique_id": UNIQUE_ID,
+            "gw_connected_topic": f"inels/connected/{MAC_ADDRESS}/gw",
+            "connected_topic": f"inels/connected/{MAC_ADDRESS}/{DT_100}/{UNIQUE_ID}",
+            "status_topic": f"inels/status/{MAC_ADDRESS}/{DT_100}/{UNIQUE_ID}",
+            "base_topic": f"{MAC_ADDRESS}/{DT_100}/{UNIQUE_ID}",
+            "switch_on_value": "07\n00\n0A\n28\n00\n",
+            "switch_off_value": "06\n00\n0A\n28\n00\n",
+            "alerts": {
+                "overflow": "06\n00\n0A\n28\n01\n",
+            },
+        },
+        "bit": {
+            "entity_type": "switch",
+            "device_type": "bit",
+            "dev_type": DT_BITS,
+            "unique_id": UNIQUE_ID,
+            "gw_connected_topic": f"inels/connected/{MAC_ADDRESS}/gw",
+            "connected_topic": f"inels/connected/{MAC_ADDRESS}/{DT_BITS}/{UNIQUE_ID}",
+            "status_topic": f"inels/status/{MAC_ADDRESS}/{DT_BITS}/{UNIQUE_ID}",
+            "base_topic": f"{MAC_ADDRESS}/{DT_BITS}/{UNIQUE_ID}",
+            "switch_on_value": b'{"state":{"000":1,"001":1}}',
+            "switch_off_value": b'{"state":{"000":0,"001":0}}',
+        },
+    }
+    return configs[request.param]
+
+
+@pytest.mark.parametrize(
+    "entity_config", ["simple_relay", "relay", "bit"], indirect=True
+)
+@pytest.mark.parametrize(
+    ("gw_available", "device_available", "expected_state"),
+    [
+        (True, False, STATE_UNAVAILABLE),
+        (False, True, STATE_UNAVAILABLE),
+        (True, True, STATE_ON),
+    ],
+)
+async def test_switch_availability(
+    hass: HomeAssistant,
+    setup_entity,
+    entity_config,
+    gw_available,
+    device_available,
+    expected_state,
+) -> None:
+    """Test switch availability and state under different gateway and device availability conditions."""
+
+    switch = await setup_entity(
```

Comment:
I'd suggest this as variable name:
```suggestion
    switch_state = await setup_entity(
```